### PR TITLE
Compatibility

### PR DESCRIPTION
**Pull request by Aymendje**
Upgrades in MyelineSeg
And now, we dont need cython anymore.
Tested on Windows 10, MacOS Sierra and Debian 8 / Fedora 23
Shoud work on all plateforms with minimal tweaking.

Only need to have python 3.5
Or miniconda 4.2 if using windows

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,53 @@
...
+
+For windows users : 
+
+    * Install miniconda *
+
+        Windows :
+
+            https://repo.continuum.io/miniconda/Miniconda2-4.2.12-Windows-x86_64.exe
+
+        Mac :
```
Comment (jcohenadad) [inline README.md]: should not be under WINDOWS section

Comment (Aymendje) [inline README.md]: README updated.
Tank you.

File: README.md [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
+
+            https://repo.continuum.io/miniconda/Miniconda2-4.2.12-MacOSX-x86_64.sh
```
Comment (jcohenadad) [inline README.md]: this downloads python-2.7, not python-3.5

**Comment by jcohenadad** [comment]
~~~
python3 main.py -c test.json
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    from core import ConfigParser
  File "/Users/julien/code/axonsegpy/axonsegpy/core/__init__.py", line 3, in <module>
    from core import ConfigParser
  File "/Users/julien/code/axonsegpy/axonsegpy/core/ConfigParser.py", line 2, in <module>
    from core.RunConfig import runAlgo
  File "/Users/julien/code/axonsegpy/axonsegpy/core/RunConfig.py", line 1, in <module>
    from algo import *
  File "/Users/julien/code/axonsegpy/axonsegpy/algo/__init__.py", line 1, in <module>
    from algo import AxonSeg
  File "/Users/julien/code/axonsegpy/axonsegpy/algo/AxonSeg.py", line 11, in <module>
    from lib import minima
ImportError: cannot import name 'minima'
~~~

**Comment by Aymendje** [comment]
Can you run 
python3 -V

NB: it's a capital V

**Comment by jcohenadad** [comment]
Python 3.5.2 :: Continuum Analytics, Inc.

**Comment by Aymendje** [comment]
Thank you 
Can you also run the command 
    uname -a

**Comment by jcohenadad** [comment]
Darwin julien-macbook.wireless.concordia.ca 14.5.0 Darwin Kernel Version 14.5.0: Thu Apr 21 20:40:54 PDT 2016; root:xnu-2782.50.3~1/RELEASE_X86_64 x86_64

**Comment by jcohenadad** [comment]
I'm wondering if the /lib shouldn't be linked to the Python package somehow?

**Comment by Aymendje** [comment]
Thank you. 
I will investigate.

**Comment by Aymendje** [comment]
Indeed 
It seems that there is a missing file. 
I will update in 30min, I currently don't have access to a computer.

File: setup.py [hunk 3]
```diff
@@ -1,48 +0,0 @@
-#!/usr/bin/env python
```
Comment (peristeri) [inline setup.py]: I recommend you keep this file. It is necessary for the packaging of the application

**Comment by Aymendje** [comment]
@jcohendad
Can you retry now?

**Comment by jcohenadad** [comment]
@Aymendje: retry what? I did a git pull, and rerun ```python main.py -c test.json``` and i got the same error

**Comment by jcohenadad** [comment]
@peristeri: during last meeting, the students told me they had difficulties to generate the binaries. I told the students that we need to be pragmatic and have a working solution for today.

**Comment by Aymendje** [comment]
@peristeri We spoke about this in last meeting. The problem is : we are using cython to generate these librairies. So they cant just be pulled from somewhere. But the cython solution need specefic librairies (openmp), and to use them, a lot of tweaking is needed.
On Mac, you need to recompile all the librairies of gcc (it takes +- 1h) with brew, then link the new gcc binary to the system path.
On Linux you need to download a few dependancies, and it depend if you are using Fedora, Debian, Arch, etc.
On Windows you need Visual Studio and a few other DLL.

As it was starting to get really complicated, we chose to give precompiled binaries in the solution, while keeping the cython source so they can be manually compiled by users who choose to do so.

**Comment by Aymendje** [comment]
@jcohenadad allright, I don't have access to the mac for now, but ill try to test it today and find the problem.

**Comment by jcohenadad** [comment]
@Aymendje: come to the lab-- we have 15 macs you can work on. @alzaia will help you

**Comment by peristeri** [comment]
@Aymendje  That is why you need an automated build process. 

You need to automate the build process in order to reproduce the installation. Time is not a good argument on doing your build manually.

**Comment by jcohenadad** [comment]
@peristeri: just for their defence: I told them time is an issue (they said it takes an hour to compile, which is a bit insane...)

**Comment by Aymendje** [comment]
In the next iteration (so for Wednesday or Thuesday)
axonseg with by default try to build the cython binaries. If it fails, it
will fallback to using the precompiled binaries in the lib folder.
This is the best arrangement we found for now.

**Comment by Aymendje** [comment]
@jcohenadad 
I found the problem
I am recompiling the dependancies and uploading it.
It will be up in an hour (I am using another computer, so I need to reinstall all the gcc dependancies...)

**Comment by Aymendje** [comment]
Can you pull and retry ?
I updated the macos lib.

**Comment by jcohenadad** [comment]
~~~
julien-macbook:~/code/axonsegpy/axonsegpy $ git pull
remote: Counting objects: 12, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 12 (delta 8), reused 9 (delta 7), pack-reused 0
Unpacking objects: 100% (12/12), done.
From github.com:neuropoly/axonsegpy
   33e33b8..1be21e5  compatibility -> origin/compatibility
Updating 33e33b8..1be21e5
Fast-forward
 axonsegpy/algo/AxonSeg.py                   |   2 +-
 axonsegpy/lib/GenMask.cpython-35m-darwin.so | Bin 0 -> 216076 bytes
 axonsegpy/lib/minima.cpython-35m-darwin.so  | Bin 0 -> 234880 bytes
 axonsegpy/test.json                         |  22 +++++++++++-----------
 4 files changed, 12 insertions(+), 12 deletions(-)
 create mode 100644 axonsegpy/lib/GenMask.cpython-35m-darwin.so
 create mode 100644 axonsegpy/lib/minima.cpython-35m-darwin.so
julien-macbook:~/code/axonsegpy/axonsegpy $ python main.py -c test.json
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    from core import ConfigParser
  File "/Users/julien/code/axonsegpy/axonsegpy/core/__init__.py", line 3, in <module>
    from core import ConfigParser
  File "/Users/julien/code/axonsegpy/axonsegpy/core/ConfigParser.py", line 2, in <module>
    from core.RunConfig import runAlgo
  File "/Users/julien/code/axonsegpy/axonsegpy/core/RunConfig.py", line 1, in <module>
    from algo import *
  File "/Users/julien/code/axonsegpy/axonsegpy/algo/__init__.py", line 1, in <module>
    from algo import AxonSeg
  File "/Users/julien/code/axonsegpy/axonsegpy/algo/AxonSeg.py", line 11, in <module>
    from lib import minima
  File "/Users/julien/code/axonsegpy/axonsegpy/lib/__init__.py", line 1, in <module>
    from lib import minima
ImportError: dlopen(/Users/julien/code/axonsegpy/axonsegpy/lib/minima.cpython-35m-darwin.so, 2): Library not loaded: /usr/local/opt/gcc/lib/gcc/6/libgomp.1.dylib
  Referenced from: /Users/julien/code/axonsegpy/axonsegpy/lib/minima.cpython-35m-darwin.so
  Reason: image not found
~~~

**Comment by Aymendje** [comment]
for this, just do :

        brew install gcc

It should take a few seconds (or a minute).
If it works, I will update the README accordingly.
Thank you.

On Mon, Apr 10, 2017 at 3:37 PM, Julien Cohen-Adad <notifications@github.com

**Comment by jcohenadad** [comment]
@Aymendje: it worked (took 30s). Waiting for your instructions for next steps

**Comment by Aymendje** [comment]
Can you retry to run

      python3 main.py

Or still same error message ?

On Mon, Apr 10, 2017 at 4:31 PM, Julien Cohen-Adad <notifications@github.com

**Comment by jcohenadad** [comment]
@Aymendje: good news
~~~
python main.py -c test.json
Applying preprocessing algorithms :

Applying axonSegmentation algorithms :

Applying axonPostProcessing algorithms :
	Calling algorithm : 'AxonSeg' with params:
		MinorMajorRatio : 0.1
		minSize : 30
		output : ../tests/SegTest/Step_2_Initial_AxonSeg.csv
		Solidity : 0.3
		input : ../tests/SegTest/20160830_CARS_Begin_07.tif
Axon seg started, params= {'MinorMajorRatio': 0.1, 'minSize': 30, 'output': '../tests/SegTest/Step_2_Initial_AxonSeg.csv', 'Solidity': 0.3, 'input': '../tests/SegTest/20160830_CARS_Begin_07.tif'}
Minima ended
RegionProps ended regions found: 776
0 %
10 %
20 %
30 %
40 %
50 %
60 %
70 %
80 %
90 %
Axon seg ended, axon found: 2
generating mask
AxonSeg ended
	Calling algorithm : 'MyelinSeg' with params:
		outputList : ../tests/SegTest/Step_2_Initial_AxonSeg.melyn.csv
		outputImage : ../tests/SegTest/Step_2_Initial_AxonSeg_mey.png
		input : ../tests/SegTest/20160830_CARS_Begin_07.tif
Axon seg started, params= {'maxSize': 1000, 'MinorMajorRatio': 0.85, 'Solidity': 0.7, 'minSize': 30}
Minima ended
RegionProps ended regions found: 776
0 %
10 %
20 %
30 %
40 %
50 %
60 %
70 %
80 %
90 %
Axon seg ended, axon found: 368
generating mask
AxonSeg ended
/Users/julien/python-3.5/lib/python3.5/site-packages/skimage/io/_io.py:132: UserWarning: ../tests/SegTest/Step_2_Initial_AxonSeg_mey.png is a low contrast image
  warn('%s is a low contrast image' % fname)
~~~

**Comment by Aymendje** [comment]
Readme updated to reflect the gcc install.
Now we will investivate the "Axon seg ended, axon found: 2" from #23

**Comment by Aymendje** [comment]
#23 should now be resolved.
We wil add the binary dependancies for python 3.6 tomorow
And also will add an automatic failover to compiling cython if the binary are not present.
(if cython is available)

**Comment by Aymendje** [comment]
Binary file for python 3.6 added.
Now works with 3.5 and 3.6
Also with any future (or precedent) version of python if the correct gcc stuff is installed.
The framework will try first to compile with cython / gcc.
If it failes, it will use the librairies.
If it fails (no compatible librairie), it will exit, saying coudnt find the librairie.

Oh, and issue #24 was already solved, it wasnt put in the test.json to show the result.
Now it is, so you can also try.

**Comment by jcohenadad** [comment]
can we merge this PR?  it's becoming a forum for all other unrelated issues

**Comment by Aymendje** [comment]
Last commit #26 shoud implement a test mode.

running main.py -t will ignore all other flags and go for "test" mode.

**Pull request merged**